File: app/src/java/java/com/example/ama/viewer/data/api/dto/GithubUserDTO.java
package com.example.ama.viewer.data.api.dto;

import android.support.annotation.Nullable;

import com.example.ama.viewer.data.entity.GithubUser;
import com.google.gson.annotations.SerializedName;

public class GithubUserDTO {
    private final static String NOT_SPECIFIED = "not specified";

    private String login;

    @Nullable
    @SerializedName("avatar_url")
    private String avatar;

    private String name;
    private String company;
    private String blog;
    private String location;
    private String email;
    private String bio;

    public GithubUserDTO(GithubUser user) {
        login = user.getLogin();
        avatar = user.getAvatar();
        name = user.getName();
        company = user.getCompany();
        blog = user.getBlog();
        location = user.getLocation();
        email = user.getEmail();
        bio = user.getBio();
    }

    public String getLogin() {
        return login == null ? NOT_SPECIFIED : login;
    }

    @Nullable
    public String getAvatar() {
        return avatar;
    }

    public String getName() {
        return name == null ? NOT_SPECIFIED : name;
    }

    public String getCompany() {
        return company == null ? NOT_SPECIFIED : company;
    }

    public String getBlog() {
        return blog == null ? NOT_SPECIFIED : blog;
    }

    public String getLocation() {
        return location == null ? NOT_SPECIFIED : location;
    }

    public String getEmail() {
        return email == null ? NOT_SPECIFIED : email;
    }

    public String getBio() {
        return bio == null ? NOT_SPECIFIED : bio;
    }
}

File: app/src/kotlin/java/com/example/ama/viewer/data/api/dto/GithubUserDTO.kt
package com.example.ama.viewer.data.api.dto

import com.example.ama.viewer.data.entity.GithubUser
import com.google.gson.annotations.SerializedName

data class GithubUserDTO(

        @SerializedName("login")
        private val _login: String?,

        @SerializedName("avatar_url")
        val avatar: String?,

        @SerializedName("name")
        private val _name: String?,

        @SerializedName("company")
        private val _company: String?,

        @SerializedName("blog")
        private val _blog: String?,

        @SerializedName("location")
        private val _location: String?,

        @SerializedName("email")
        private val _email: String?,

        @SerializedName("bio")
        private val _bio: String?) {

    constructor(user: GithubUser) : this(
            user.login,
            user.avatar,
            user.name,
            user.company,
            user.blog,
            user.location,
            user.email,
            user.bio)

    val login: String
        get() = _login ?: NOT_SPECIFIED

    val name: String
        get() = _name ?: NOT_SPECIFIED

    val company: String
        get() = _company ?: NOT_SPECIFIED

    val location: String
        get() = _location ?: NOT_SPECIFIED

    val blog: String
        get() = _blog ?: NOT_SPECIFIED

    val email: String
        get() = _email ?: NOT_SPECIFIED

    val bio: String
        get() = _bio ?: NOT_SPECIFIED

    companion object {
        const val NOT_SPECIFIED = "Not specified"
    }
}